src/Refactoring-UI/RBRenameVariableDriver.class.st

```diff
@@ -0,0 +1,131 @@
+"
+I'm an abstract model for user interaction for the rename variables refactoring.
+"
+Class {
+	#name : 'RBRenameVariableDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'oldName',
+		'newName',
+		'lastName',
+		'class',
+		'shouldEscape'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'execution' }
+RBRenameVariableDriver >> applyChanges [
+	"I use the preview presenter to perform changes, but I do not open it"
+	(self previewWithChanges: self changes) performChanges
```

Comment:
Does it make sense to use refactoring here `refactoring performChanges`?

Reply:
Yes it does, but if I use `refactoring performChanges`, changes will always be performed. I pass by the Preview Presenter because if the presenter is an instance of `StRefactoringPreviewPresenterMock` the changes won't be performed,  whereas if the presenter is an instance of `StRefactoringPreviewPresenter`, the changes will be performed

Reply:
I did not look at the first driver tests I did but I wonder why I did not get the same problems.

Reply:
I just read it and I do not see why you get this problem. 
Did you check the RemoveClassDriver?
Because call the applyChanges calls the presenter and there is no problem there, since it invokes the presenter
and do not open it and this is not a problem. We are not testing the previewPresenter here but the driver. 
So I do not care that it does not open I care that I can validate that a certain number and kind of changes are done. Does it help you?

```
runRefactoring
	
	self configureRefactoring.
	refactoring checkApplicabilityPreconditions.
	self setBreakingChangesPreconditions.
	haveNoReferences check & emptyClasses check & noSubclasses check
			ifTrue: [ self applyChanges ]
			ifFalse: [ self handleBreakingChanges ]
```

Reply:
So I do not understand why you have to redefine applyChanges
In the superclass it is defined as 
```
applyChanges

	self openPreviewWithChanges: self changes
```

And to me it does the job.
So why do you redefine it.

Reply:
@carolahp did you see this comment?